metabase/metabase
# fix static visualizations dashcard settings merging

Closes https://github.com/metabase/metabase/issues/47495

### Description

This PR fixes broken static charts in the following case:
- A native question was added to a dashboard
- Dashcard viz settings were changed
- Then the native query got updated with columns renaming

The static viz renderer incorrectly handled dashcard settings and this PR makes it the same as in the dynamic viz.

### How to verify

- Create a question
```
select 'foo' x, 10 y
union all select 'bar', 20
```

- Add on a dashboard
- Update any dashcard viz setting
- Go to the question and update query to 
```
select 'foo' x, 10 y1
union all select 'bar', 20
```
- Go to the dashboard and try sending a subscription
- Ensure it does not break

### Checklist
